internal/graph/iterator_stream.go

```diff
@@ -0,0 +1,133 @@
+package graph
+
+import (
+	"context"
+	"slices"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/storage"
+)
+
+type iteratorStream struct {
```

Comment:
Could we add more methods to this struct so that callers _do not_ have to rely on internal fields? For example:
```
func (s *iteratorStream) Head(ctx context.Context) (*openfgav1.TupleKey, error) {
	return s.buffer.Head(ctx)
}
```

Then, in `check_fast_path.go`, we could replace line 199

```go
v, err := stream.buffer.Head(ctx)
```

with 

```go
v, err := stream.Head(ctx)
```